creator-node/src/snapbackSM.js

```diff
@@ -47,63 +61,387 @@ class SnapbackSM {
+    this.log(`Issued ${numSyncsIssued} sync ops`)
+    this.log(`------------------END Process SnapbackSM Operation, slice ${previousModuloSlice} ------------------`)
+  }
+
+  // Track an ongoing sync operation to a secondary
+  async monitorSecondarySync (syncWallet, primaryClockValue, secondaryUrl) {
+    let startTime = Date.now()
+    // Monitor the sync status
+    let syncMonitoringRequestParameters = {
+      method: 'get',
+      baseURL: secondaryUrl,
+      url: `/sync_status/${syncWallet}`,
+      responseType: 'json'
+    }
+    let syncAttemptCompleted = false
+    while (!syncAttemptCompleted) {
       try {
-        await this.processStateMachineOperation(job)
+        let syncMonitoringResp = await axios(syncMonitoringRequestParameters)
+        let respData = syncMonitoringResp.data.data
+        this.log(`processSync ${syncWallet} secondary response: ${JSON.stringify(respData)}`)
+        // A success response does not necessarily guarantee completion, validate response data to confirm
+        if (respData.clockValue === primaryClockValue) {
+          syncAttemptCompleted = true
+          this.log(`processSync ${syncWallet} clockValue from secondary:${respData.clockValue}, primary:${primaryClockValue}`)
+        }
       } catch (e) {
-        logger.info(`Error processing ${e}`)
-      } finally {
-        this.log(`DEV MODE next job in ${DevDelayInMS}ms at ${new Date(Date.now() + DevDelayInMS)}`)
-        await utils.timeout(DevDelayInMS)
-        this.stateMachineQueue.add({ startTime: Date.now() })
-        done()
+        this.log(`processSync ${syncWallet} error querying sync_status: ${e}`)
+      }
+      if (Date.now() - startTime > MaxSyncMonitoringDurationInMs) {
+        this.log(`ERROR: processSync ${syncWallet} timeout for ${syncWallet}`)
+        syncAttemptCompleted = true
       }
+      // 1s delay between retries
+      await utils.timeout(1000)
+    }
+    let duration = Date.now() - startTime
+    if (!syncAttemptCompleted) {
+      this.log(`Sync for ${syncWallet} at ${secondaryUrl} completed in ${duration}ms`)
+    }
+  }
+
+  // Main sync queue job
+  async processSyncOperation (job) {
+    const syncRequestParameters = job.data.syncRequestParameters
+    let isValidSyncJobData = (
+      ('baseURL' in syncRequestParameters) &&
+      ('url' in syncRequestParameters) &&
+      ('method' in syncRequestParameters) &&
+      ('data' in syncRequestParameters)
+    )
+    if (!isValidSyncJobData) {
+      logger.error(`Invalid sync data found`)
+      logger.error(job.data)
+      return
+    }
+    const syncWallet = syncRequestParameters.data.wallet[0]
+    const primaryClockValue = await this.getUserPrimaryClockValue(syncWallet)
+    const secondaryUrl = syncRequestParameters.baseURL
+    this.log(`------------------Process SYNC | User ${syncWallet} | Target: ${secondaryUrl} ------------------`)
+
+    // Issue sync request to secondary
+    await axios(syncRequestParameters)
```

Comment:
does this block on the axios returning before it goes onto monitorSecondarySync?

Reply:
i think that's what await does right?